gateway/src/main/scala/com/improving/app/gateway/infrastructure/routes/DemoScenarioGatewayRoutes.scala

```diff
@@ -59,78 +68,129 @@ trait DemoScenarioGatewayRoutes extends ErrorAccumulatingCirceSupport with Stric
+                      .zip(Random.shuffle(addresses))
+                      .zip(Random.shuffle(cityStates))
+                      .take(parsed.numTenants)
+                      .map {
+                        case (((first, last), address), Seq(city, state)) =>
+                          tenantHandler
+                            .establishTenant(
+                              EstablishTenant(
+                                Some(TenantId(UUID.randomUUID().toString)),
+                                creatingMemberId,
+                                Some(
+                                  EditableTenantInfo(
+                                    Some("Demo-" + LocalDateTime.now().toString),
+                                    Some(
+                                      EditableContact(
+                                        Some(first),
+                                        Some(last),
+                                        Some(s"$first.$last@orgorg.com"),
+                                        Some(genPhoneNumber),
+                                        Some(first.take(1) + last)
+                                      )
+                                    ),
+                                    Some(
+                                      EditableAddress(
+                                        line1 = Some(address),
+                                        line2 = None,
+                                        city = Some(city),
+                                        stateProvince = Some(state),
+                                        country = Some("Fakaria"),
+                                        postalCode = Some(
+                                          PostalCodeMessageImpl(
+                                            UsPostalCodeImpl(genPostalCode)
+                                          )
+                                        )
+                                      )
+                                    ),
+                                    Some(TenantOrganizationList(Seq(orgId)))
                                   )
                                 )
-                              ),
-                              Some(TenantOrganizationList(Seq(orgId)))
+                              )
                             )
+                        case (((_, _), _), _) =>
+                          Future.failed(new InternalServerErrorException("Could not generate all appropriate data"))
+                      }
+                      .map(_.flatMap { tenantEstablished: TenantEstablished =>
+                        tenantHandler
+                          .activateTenant(ActivateTenant(tenantEstablished.tenantId, creatingMemberId))
+                          .map(_ => tenantEstablished.toTenant)
+                      })
+                  )
+
+              val orgResponse: Future[Organization] = tenantResponses.flatMap { tenants =>
+                (Random
```

Comment:
What's the thought behind randomizing these lists?

Reply:
It will make the demo look different every time we set up. That will help with making it feel more real to users